src/AzSK.ADO/Framework/BugLog/AutoBugLog.ps1

```diff
@@ -430,39 +463,53 @@ class AutoBugLog {
     }
 
     #function to search for existing bugs based on the hash
-    hidden [object] GetWorkItemByHash([string] $hash, [string] $ProjectName) {
-		
-        $url = "https://{0}.almsearch.visualstudio.com/{1}/_apis/search/workItemQueryResults?api-version=5.1-preview" -f $this.OrganizationName, $ProjectName
-
-        #TODO: validate set to allow only two values : ReactiveOldBug and CreateNewBug
-        #check for ResolvedBugBehaviour in control settings
-        #takeResults is used to fetch number of workitems to be return. At caller side of this method we are checking if return greter then 0, then manage work item else add new.
-        if ($this.ControlSettings.BugLogging.ResolvedBugLogBehaviour -ne "ReactiveOldBug") {
-            #new bug is to be logged for every resolved bug, hence search for only new/active bug
-            $body = '{"searchText":"{0}","skipResults":0,"takeResults":2,"sortOptions":[],"summarizedHitCountsNeeded":true,"searchFilters":{"Projects":["{1}"],"Work Item Types":["Bug"],"States":["Active","New"]},"filters":[],"includeSuggestions":false}' | ConvertFrom-Json
-        }
-        else {
-            #resolved bug needs to be reactivated, hence search for new/active/resolved bugs
-            $body = '{"searchText":"{0}","skipResults":0,"takeResults":2,"sortOptions":[],"summarizedHitCountsNeeded":true,"searchFilters":{"Projects":["{1}"],"Work Item Types":["Bug"],"States":["Active","New","Resolved"]},"filters":[],"includeSuggestions":false}' | ConvertFrom-Json
+    hidden [object] GetWorkItemByHash([string] $hash, [string] $ProjectName) 
+    {
+        #TODO: 
+        #$this.ScanSource = "CA";
+        if ($this.UseAzureStorageAccount -and $this.ScanSource -eq "CA") 
+        {
+            return $this.BugLogCheckerObj.GetWorkItemByHashAzureTable($hash, $ProjectName, $this.ControlSettings.BugLogging.ResolvedBugLogBehaviour);
```

Comment:
ResolvedBugLogBehaviour - do we need this?

Reply:
removed